plugins/datastreamer_plugin/utils/dynamic_types_utils.cpp

```diff
@@ -38,717 +43,149 @@ namespace plotjuggler {
 namespace utils {
 
 std::vector<std::string> get_introspection_type_names(
-        const TypeIntrospectionCollection& numeric_type_names)
+        const TypeIntrospectionNumericStruct& numeric_type_names)
 {
     std::vector<std::string> type_names;
     for (const auto& type_name : numeric_type_names)
     {
-        type_names.push_back(std::get<types::DatumLabel>(type_name));
+        type_names.push_back(type_name.first);
     }
     return type_names;
 }
```

Comment:
You could use templates instead of repeating the same implementation twice:

```suggestion
template<typename T>
std::vector<std::string> get_introspection_type_names(const T& type_names_struct)
{
    std::vector<std::string> type_names;
    for (const auto& type_name : type_names_struct)
    {
        type_names.push_back(type_name.first);
    }
    return type_names;
}
```